client/map_verifier.go

```diff
@@ -111,42 +110,15 @@ func (m *MapVerifier) VerifyMapLeavesResponse(indexes [][]byte, revision int64,
 		return nil, status.Errorf(codes.Internal, "got map revision %v, want %v", mapRoot.Revision, revision)
 	}
 
-	proofs := make(chan *trillian.MapLeafInclusion)
-	done := make(chan struct{})
-	defer close(done)
-	go func() {
-		defer close(proofs)
-		for _, p := range resp.MapLeafInclusion {
-			select {
-			case proofs <- p:
-			case <-done:
-				return
-			}
-		}
-	}()
-	errors := make(chan error)
-	go func() {
-		defer close(errors)
-		var wg sync.WaitGroup
-		defer wg.Wait()
-		for w := 0; w < runtime.NumCPU(); w++ {
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-				for p := range proofs {
-					select {
-					case errors <- m.VerifyMapLeafInclusionHash(mapRoot.RootHash, p):
-					case <-done:
-						return
-					}
-				}
-			}()
-		}
-	}()
-	for err := range errors {
-		if err != nil {
-			return nil, status.Errorf(status.Code(err), "map: VerifyMapLeafInclusion(): %v", err)
-		}
+	var g errgroup.Group
+	for _, p := range resp.MapLeafInclusion {
+		p := p
+		g.Go(func() error {
```

Comment:
This code always starts all the verifications, whereas the previous one could be canceled if one failed? Is this critical? I guess, we should care mostly about the success case.

Reply:
Thanks for catching that. The documentation said it would cancel things, but it only cancels when called from `WithContext` I'll look into this more tomorrow.

Reply:
Thinking about this more... 
Explicit cancels are useful when one is in the middle of a pipeline and one can use
```go
select {
data <- inputpipe:
    process
<- ctx.Done():
    exit pipeline
}
```

Because this instantiation does not have a pipeline, I don't think checking ctx.Done() is useful since it will only create an unreliable race condition. 

Your observation that this is mostly about the success case is also correct.